### ISSUE TITLE
Error: "The targeted input element must be the active input element" when switching focus between TextFormField

### ISSUE THREAD

**Issue by bachcmd** [flutter/flutter#156842]
### Steps to reproduce

1. Create a form with multiple TextFormField widgets.
2. Use the mouse or keyboard to quickly (very quickly) switch focus between the fields.
3. Error appears in browser debug console

### Expected results

Switching focus between TextFormField widgets should work smoothly without any error messages.

### Actual results

An error message appears when switching focus quickly, disrupting the user experience.

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![image](https://github.com/user-attachments/assets/bd48bea7-15f0-458e-85b5-0db12dc30e16)

</details>

### Logs

<details open><summary>Logs</summary>

```console

TextFormField 3 lost focus
TextFormField 1 lost focus
TextFormField 2 lost focus
DartError: Assertion failed: org-dartlang-sdk:///lib/_engine/engine/pointer_binding/event_position_helper.dart:64:10
targetElement == domElement
"The targeted input element must be the active input element"
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 296:3     throw_
errors.dart:296
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 29:3      assertFailed
errors.dart:29
lib/_engine/engine/pointer_binding/event_position_helper.dart 64:37             _computeOffsetForInputs
event_position_helper.dart:64
lib/_engine/engine/pointer_binding/event_position_helper.dart 32:14             computeEventOffsetToTarget
event_position_helper.dart:32
lib/_engine/engine/pointer_binding.dart 1078:30                                 [_convertEventsToPointerData]
pointer_binding.dart:1078
lib/_engine/engine/pointer_binding.dart 1017:9                                  <fn>
pointer_binding.dart:1017
lib/_engine/engine/pointer_binding.dart 941:7                                   <fn>
pointer_binding.dart:941
lib/_engine/engine/pointer_binding.dart 534:9                                   loggedHandler
pointer_binding.dart:534
dart-sdk/lib/_internal/js_dev_runtime/patch/js_allow_interop_patch.dart 188:27  _callDartFunctionFast1
js_allow_interop_patch.dart:188
TextFormField 1 lost focus
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel master, 3.24.0-1.0.pre.114, on Microsoft Windows [Version 10.0.14393], locale en-GB)
    • Flutter version 3.24.0-1.0.pre.114 on channel master at C:\flutter\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 82b63ff27d (3 months ago), 2024-07-14 05:07:36 -0400
    • Engine revision a921637f16
    • Dart version 3.6.0 (build 3.6.0-36.0.dev)
    • DevTools version 2.37.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/windows-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/windows-android-setup for detailed instructions).

[√] VS Code (version 1.94.2)
    • VS Code at C:\Users\BlueSky\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.98.0

[√] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.14393]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 128.0.6613.113

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 3 categories.
```

</details>

**Comment by bachcmd** [comment]
Sample code here:
```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Form Input Example',
      home: MyForm(),
    );
  }
}

class MyForm extends StatefulWidget {
  @override
  _MyFormState createState() => _MyFormState();
}

class _MyFormState extends State<MyForm> {
  final FocusNode _focusNode1 = FocusNode();
  final FocusNode _focusNode2 = FocusNode();
  final FocusNode _focusNode3 = FocusNode();

  @override
  void initState() {
    super.initState();
    
    _focusNode1.addListener(() {
      if (!_focusNode1.hasFocus) {
        print('TextFormField 1 lost focus');
      }
    });

    _focusNode2.addListener(() {
      if (!_focusNode2.hasFocus) {
        print('TextFormField 2 lost focus');
      }
    });

    _focusNode3.addListener(() {
      if (!_focusNode3.hasFocus) {
        print('TextFormField 3 lost focus');
      }
    });
  }

  @override
  void dispose() {
    _focusNode1.dispose();
    _focusNode2.dispose();
    _focusNode3.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Form Input Example'),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: FocusTraversalGroup(
          child: Form(
            child: Column(
              children: [
                TextFormField(
                  focusNode: _focusNode1,
                  decoration: InputDecoration(labelText: 'Field 1'),
                ),
                SizedBox(height: 16),
                TextFormField(
                  focusNode: _focusNode2,
                  decoration: InputDecoration(labelText: 'Field 2'),
                ),
                SizedBox(height: 16),
                TextFormField(
                  focusNode: _focusNode3,
                  decoration: InputDecoration(labelText: 'Field 3'),
                ),
                SizedBox(height: 16),
                ElevatedButton(
                  onPressed: () {
                    print('Submit button pressed');
                  },
                  child: Text('Submit'),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
```

**Comment by huycozy** [comment]
Thanks for your report. Reproduced this on Web app. No error appeared on the MacOS app with the same steps.

Flutter versions:
```console
Channel stable, 3.24.3
Channel master, 3.27.0-1.0.pre.64
```

**Comment by bachcmd** [comment]
https://github.com/user-attachments/assets/4f286f55-93ee-4409-8723-d514cbbebfe5

Note that the error occurs when switching focus between TextFormField fields very quickly

**Comment by justinmc** [comment]
Seems like this is likely something web-specific, but if it's better off being handled by the text input team feel free to let me know.

**Comment by rafhaelamarques** [comment]
I was getting the same error and same scenario only Web. The problem was on the _index.html_. Depending your project, just recreate the web folder must help. 

So delete the web folder and run `flutter create --platforms web .` 

The _index.html_ will be recreate with the proper updated configurations.

Since my project is a production app with custom index.html, I just adjusted some points on that file. In my case a script was out of scope, it was the `<script src="flutter.js" defer></script>` being calling out of _<body>_ tag. This solved the problem like a charm. 

But I went further and updated the index.file with the new [Flutter Web App Initialization](https://docs.flutter.dev/platform-integration/web/initialization) which replace the _flutter.js_ to _flutter_bootstrap.js_.

Here's a sample of the body tag from my index.html working now:

```
<body>
  <script>
    window.addEventListener('load', function(ev) {
      // Adds the flutter initialization JS code
      {{flutter_js}}
      {{flutter_build_config}}

      // Download main.dart.js
      _flutter.loader.load({
        serviceWorker: {
          serviceWorkerVersion: "{{flutter_service_worker_version}}",
        },
        onEntrypointLoaded: function(engineInitializer) {
          engineInitializer.initializeEngine().then(function(appRunner) {
            appRunner.runApp();
          });
        }
      });
    });
  </script>
</body>
```
Hope it helps anyone facing this issue.

**Comment by bachcmd** [comment]
I tried deleting the web folder and recreating it but the error still appears. When I recreate it with the xxxxx command, the body tag content is: 
```
<body>
    <script src="flutter_bootstrap.js" defer></script>
</body>
```

**Comment by ditman** [comment]
Thanks for reporting! The bug points to this part of the engine:

* https://github.com/flutter/engine/blame/bcb281cde579a3072adc04a8ba2d0709cd815fb3/lib/web_ui/lib/src/engine/pointer_binding/event_position_helper.dart#L70

(We use that helper to transform browser coordinates to app coordinates)

This is a race condition between our rendering (when adding an input, there's some DOM operations that need to happen) and the mouse click/tap handlers.

**Comment by mstestlp** [comment]
不需要快速切换,直接切换2个输入框的焦点来回2次,然后鼠标移动到页面的最顶部100%复现

No need for quick switching, simply switch the focus of two input boxes back and forth twice, and then move the mouse to the top of the page to reproduce 100%

**Comment by crich04** [comment]
I am having same issue.  On iOS, I cannot access the fields that exhibit this issue.

**Comment by gardeninghouse** [comment]
I also get the same problem after updating to 3.27.0. Just typing something in the text field gets the same error.

**Comment by itforgeuk** [comment]
I also get the same problem for web platform after updating to 3.27.0. I tried focusnodenotes.requestFocus(); onTap but didn't help

**Comment by eduardofcr** [comment]
I also get the same problem just after updating to 3.27.0

**Comment by Waxo** [comment]
Same problem since 3.27.0 update, not platform related. Got it when you focus a TextField.

Console is unusable right now even with debugger.

**Comment by famartinez80** [comment]
I also get the same problem just after updating to 3.27.0

**Comment by Eng-Mohamed-Elsayed** [comment]
I also get the same problem just after updating to 3.27.0

**Comment by Kemerd** [comment]
You can add this to your index.html to ignore this until it gets fixed

```
  <script>
    // Prevent breaking on pointer binding assertions
    window.addEventListener('error', function (e) {
      if (e.message.includes('targetElement == domElement') ||
        e.message.includes('The targeted input element must be the active input element')) {
        e.preventDefault();
        e.stopPropagation();
        console.warn('Suppressed pointer binding assertion:', e.message);
        return true;
      }
    });

    // Additional error handling for development
    if (!window.console) window.console = {};
    if (!window.console.error) window.console.error = function () { };
    window.console.error = (function (old) {
      return function (msg) {
        if (msg && typeof msg === 'string' &&
          (msg.includes('targetElement == domElement') ||
            msg.includes('The targeted input element must be the active input element'))) {
          return;
        }
        return old.apply(this, arguments);
      };
    })(window.console.error);
  </script>
```

**Comment by fruts** [comment]
This issue ruin web development and sentry integrations

**Comment by otopba** [comment]
Same here 3.27.0

**Comment by Vitor1703** [comment]
Same problem after updating to 3.27.0

**Comment by WingCH** [comment]
Same here 3.27.0

**Comment by cephyn** [comment]
I will add to the chorus, getting this on my login page after upgrading to 3.27.0 - the script fix from @Kemerd did not work for me.

**Comment by EliaTolin** [comment]
> Same issue in 3.27.0 for me
> 
> You can add this to your index.html to ignore this until it gets fixed. Or disable breakpoints in VS code
> 
> ```
>   <script>
>     // Prevent breaking on pointer binding assertions
>     window.addEventListener('error', function (e) {
>       if (e.message.includes('targetElement == domElement') ||
>         e.message.includes('The targeted input element must be the active input element')) {
>         e.preventDefault();
>         e.stopPropagation();
>         console.warn('Suppressed pointer binding assertion:', e.message);
>         return true;
>       }
>     });
> 
>     // Additional error handling for development
>     if (!window.console) window.console = {};
>     if (!window.console.error) window.console.error = function () { };
>     window.console.error = (function (old) {
>       return function (msg) {
>         if (msg && typeof msg === 'string' &&
>           (msg.includes('targetElement == domElement') ||
>             msg.includes('The targeted input element must be the active input element'))) {
>           return;
>         }
>         return old.apply(this, arguments);
>       };
>     })(window.console.error);
>   </script>
> ```

This work-around it works but insert this script in body tag.

**Comment by juppdes** [comment]
It's a shame that the Flutter team apparently doesn't pay due attention to the Web platform.. :/

**Comment by 93jose93** [comment]
I get the error in the input just like on the web, when you hit the input it freezes, but if I restart, the error is shown, it is the following:
DartError: Assertion failed: org-dartlang-sdk:///lib/_engine/engine/pointer_binding/event_position_helper.dart:70:10
targetElement == domElement
"The targeted input element must be the active input element"
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 288:3     throw_
errors.dart:288
dart-sdk/lib/_internal/js_dev_runtime/private/profile.dart 110:39               assertFailed
profile.dart:110
dart-sdk/lib/_internal/js_dev_runtime/private/profile.dart 110:39               _computeOffsetForInputs
profile.dart:110
lib/_engine/engine/pointer_binding/event_position_helper.dart 38:14             computeEventOffsetToTarget
event_position_helper.dart:38
lib/_engine/engine/pointer_binding.dart 1088:30                                 [_convertEventsToPointerData]
pointer_binding.dart:1088
lib/_engine/engine/pointer_binding.dart 1016:9                                  <fn>
pointer_binding.dart:1016
lib/_engine/engine/pointer_binding.dart 948:7                                   <fn>
pointer_binding.dart:948
lib/_engine/engine/pointer_binding.dart 541:9                                   loggedHandler
pointer_binding.dart:541
dart-sdk/lib/_internal/js_dev_runtime/patch/js_allow_interop_patch.dart 212:27  _callDartFunctionFast1
js_allow_interop_patch.dart:212
Restarted application in 95ms.

**Comment by EliaTolin** [comment]
But this bug is unbelievable. 
It means that it was not tested once on the Web before release! 
It is really blatant as an error. Do you have any other explanations?

**Comment by Macacoazul01** [comment]
A cherry-pick request was made in https://github.com/flutter/engine/pull/57246.

**Comment by EliaTolin** [comment]
If i try to render with WebAssembly (--wasm), I don’t see any issues, and it works perfectly.

**Comment by eduardofcr** [comment]
after upgrade Chrome to 131.0.6778.204 and flutter to 3.27.1, the issue was solved.
It looks like it was a double bug...
the issue does not happens in Edge 131.0.2903.99, but I did not tried in Safari...

**Comment by j-fbriere** [comment]
Just made the test of upgrading Chrome with Flutter stable.
Upgrading to Chrome 131.0.6778.204 (in my case 131.0.6778.205) and Flutter stable to 3.27.1 does **NOT** solve the problem.
The error: "The targeted input element must be the active input element" still appears.
The test was made on Windows 10 Pro.

**Comment by Ali-Toosi** [comment]
Not sure if it's mentioned elsewhere but to clarify, the issue seems to happen when the mouse hovers over an active text field. If you focus on it by pressing tab or if it has `autofocus`, that doesn't break the assertion. Only when the mouse goes over it.

**Comment by jezell** [comment]
Also seeing this. Seems to prevent selection of single characters with cursor. Also the <form> elements inserted by the engine stay in the dom.

**Comment by JamesRaphaelJRC** [comment]
Same here, also got same error

**Comment by aramnhammer** [comment]
Seeing this as well:

chrome version: `131.0.6778.205`

flutter:
```
Flutter 3.27.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 17025dd882 (8 days ago) • 2024-12-17 03:23:09 +0900
Engine • revision cb4b5fff73
Tools • Dart 3.6.0 • DevTools 2.40.2
```

traceback after pressing continue in the debugger:
```
DartError: Assertion failed: org-dartlang-sdk:///lib/_engine/engine/pointer_binding/event_position_helper.dart:70:10
targetElement == domElement
"The targeted input element must be the active input element"
```

**Comment by brunoraz** [comment]
Is anyone addressing this problem? I think Flutter is a good framework, but these setbacks don't help. At first, I thought it was an issue with my own framework and spent a lot of time on this problem, only to discover this topic

**Comment by avatarKaran** [comment]
got this error after updating flutter

**Comment by torres-semernin** [comment]
Chiming in, i am also getting this error. downgrading flutter versions - the error does not happen. downgrading flutter version until a fix is published.

**Comment by torres-semernin** [comment]
> after upgrade Chrome to 131.0.6778.204 and flutter to 3.27.1, the issue was solved. It looks like it was a double bug... the issue does not happens in Edge 131.0.2903.99, but I did not tried in Safari...

i also have Version 131.0.6778.205 (Official Build) (arm64) and 3.27.1 but the problem persists

**Comment by Rajual** [comment]
Hi. I had the same problem. So I upgrade to 3.27.1 and delete the build folder in mi flutter project. When run debug again this created the build folder newly and the problem was resolve.

**Comment by MirceaX2Mobile** [comment]
This happens to me also on 3.27.0

**Comment by Prashant4900** [comment]
I also facing same issue in chrome latest version [Version 131.0.6778.205 (Official Build) (arm64)] but for some reason it's work perfectly in brave.

**Comment by quentin7b** [comment]
Same there on Chrome with 3.27.1

**Comment by zeroSal** [comment]
Same on Chrome and Flutter 3.27.1

**Comment by StultusRex** [comment]
same issue with 3.27.1

**Comment by ac-arconomy** [comment]
I have the same issue using the https://pub.dev/packages/searchable_listview widget.

Unable to to type into the search field without causing the error.

DartError: Assertion failed: org-dartlang-sdk:///lib/_engine/engine/pointer_binding/event_position_helper.dart:70:10
targetElement == domElement
"The targeted input element must be the active input element"
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 288:3     throw_
errors.dart:288
dart-sdk/lib/_internal/js_dev_runtime/private/profile.dart 110:39               assertFailed
profile.dart:110
dart-sdk/lib/_internal/js_dev_runtime/private/profile.dart 110:39               _computeOffsetForInputs
profile.dart:110
lib/_engine/engine/pointer_binding/event_position_helper.dart 38:14             computeEventOffsetToTarget
event_position_helper.dart:38
lib/_engine/engine/pointer_binding.dart 1088:30                                 [_convertEventsToPointerData]
pointer_binding.dart:1088
lib/_engine/engine/pointer_binding.dart 1016:9                                  <fn>
pointer_binding.dart:1016
lib/_engine/engine/pointer_binding.dart 948:7                                   <fn>
pointer_binding.dart:948
lib/_engine/engine/pointer_binding.dart 541:9                                   loggedHandler
pointer_binding.dart:541
dart-sdk/lib/_internal/js_dev_runtime/patch/js_allow_interop_patch.dart 212:27  _callDartFunctionFast1

thanks

**Comment by avatarKaran** [comment]
completely crashes the app. is there any work around? or what flutter version should i downgrade to? (I had updated my flutter after a very long time)

**Comment by j-fbriere** [comment]
> completely crashes the app. is there any work around? or what flutter version should i downgrade to? (I had updated my flutter after a very long time)

You should downgrade to Flutter stable 3.24.5.

**Comment by Ali-Toosi** [comment]
> completely crashes the app. is there any work around?
Note that it doesn't crash the production app. It's an assertion and only crashes in debug mode. If you can work around it in development, your app will still be fine in production. Or at least I haven't seen any issues being reported about this in release.

**Comment by Quantentunnel** [comment]
But I'm in the middle of development and I normally always work in debug mode.
I also have this problem with my web application under Flutter 3.27.1. With my many text fields, my application is currently unusable.
I don't want to downgrade because I had to make various adjustments for version 3.27.1, which I would otherwise have to undo.

If I can help in any way, please let me know.

**Comment by kyam28** [comment]
When there are multiple TextFields and you switch back and forth between them, issues can occur, but setting up FocusNodes can prevent this.

**Comment by mohammadhassan1497** [comment]
Same issue here, in version **3.27.1** and I updated my flutter to **3.28.0-0.1.pre** but still the same issue.

**Comment by ReinisSprogis** [comment]
Hard to debug app as it stops on this exception all the time.

**Comment by csi2345** [comment]
I also faced the same issue after upgrading sdk.

------------
Dart SDK version: 3.6.0 (stable) (Thu Dec 5 07:46:24 2024 -0800) on "windows_x64"

Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.27.1, on Microsoft Windows [Version 10.0.22631.4460], locale ko-KR)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/to/windows-android-setup for more details.
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.0)
[√] Android Studio (version 2024.1)
[√] VS Code, 64-bit edition (version 1.96.2)
[√] Connected device (3 available)
[√] Network resources

! Doctor found issues in 1 category.

**Comment by felipeduarte26** [comment]
same problem here, I use version 3.27.1 in my flutter web project.

**Comment by pitris90** [comment]
i encountered same issue as well.. downgrading to 3.24.5 fixed this issue as someone stated above

**Comment by cheymos** [comment]
+

**Comment by antiv** [comment]
same problem here, after update to 3.27.1

**Comment by KevinSinSH** [comment]
same problem here

**Comment by HeySitam** [comment]
Getting same issue with flutter stable channel 3.27.0

**Comment by dvnelson72** [comment]
same same.

**Comment by JayJayM** [comment]
Same error...

**Comment by BenjaminPrinc** [comment]
Same error, but if you use the tab-key to navigate through the application it works without issues, for me.

**Comment by DominicOrga** [comment]
I encountered this same issue after upgrading to 3.27.

**Comment by AndrewLuecke** [comment]
I can confirm same error on Mac OS

andrew@AndrewsacStudio hikefinder % flutter doctor -v
[✓] Flutter (Channel stable, 3.27.1, on macOS 15.2 24C101 darwin-arm64, locale en-AU)
    • Flutter version 3.27.1 on channel stable at /Users/andrew/Github/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 17025dd882 (3 weeks ago), 2024-12-17 03:23:09 +0900
    • Engine revision cb4b5fff73
    • Dart version 3.6.0
    • DevTools version 2.40.2

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/andrew/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

[✓] VS Code (version 1.96.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.102.0

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 15.2 24C101 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 15.2 24C101 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 131.0.6778.205

[✓] Network resources
    • All expected network resources are available.

• No issues found!

**Comment by jaysongo** [comment]
Same issue for me. MacOs + Flutter 3.27.1 + Chrome 131.0.6778.205 + flutter create ... -web
It's also not just TextFormField, it also happens with TextField.
Upvoting.

**Comment by jezell** [comment]
@kevmoo I'm not sure P2 is the right label for textboxes being broken in a stable release.

**Comment by kevmoo** [comment]
We're certain this is already fixed. It was caused by a change in chrome. Over the holidays and the monorepo merge we need to figure out our cherry pick story. Thanks for your patience.

**Comment by jezell** [comment]
@kevmoo thanks for the update

**Comment by samariotorres** [comment]
can confirm this issue is still occurring on 3.27.1

**Comment by kevmoo** [comment]
Here's the cherry-pick request if folks want to follow along: https://github.com/flutter/engine/pull/57246

**Comment by kevmoo** [comment]
We're waiting on the release engineering team to pick up and merge https://github.com/flutter/engine/pull/57246. Thanks for your patience.

**Comment by kevmoo** [comment]
Closing as duplicate of https://github.com/flutter/flutter/issues/159804

(Yes, this is the older bug. Sorry. Please follow along with https://github.com/flutter/engine/pull/57246 for updates)

**Comment by Quantentunnel** [comment]
Kevmoo wrote 3 days ago in this thread that it was caused by a change in chrome. So I tried to use Firefox and Brave instead and then it really seems to cause no problem when using the Textfields with these browsers. 
For now, that would be a workaround for me. But I still very much hope that the problem will be solved soon.

**Comment by lukaqll** [comment]
same problem here. I created a new project today with flutter v3.27.1 (macOS).
Any predictions for publishing the bug fix?

**Comment by zoheirallafzadeh** [comment]
My problem is now solved in:  Flutter 3.28.0-2.0.pre.38665 • channel master

**Comment by houssam15** [comment]
this version Flutter 3.28.0-2.0.pre.38665 • channel master solve problem

**Comment by jaysongo** [comment]
Flutter 3.27.0-0.2.pre also fixes it. Thank you!

For those interested in pulling this version down, you can refer to this to help you switch to this version of flutter until this fix drops on stable.
https://docs.flutter.dev/release/upgrade#switching-to-a-specific-flutter-version

**Comment by huycozy** [comment]
The fix seems to have been landed on stable 3.27.2: https://github.com/flutter/flutter/blob/master/CHANGELOG.md#3272
